pkg/client/claimaccess.go

```diff
@@ -0,0 +1,84 @@
+package client
+
+import (
+	"fmt"
+
+	"github.com/storacha/go-libstoracha/capabilities/access"
+	captypes "github.com/storacha/go-libstoracha/capabilities/types"
+	uclient "github.com/storacha/go-ucanto/client"
+	udelegation "github.com/storacha/go-ucanto/core/delegation"
+	"github.com/storacha/go-ucanto/core/invocation"
+	"github.com/storacha/go-ucanto/core/receipt"
+	"github.com/storacha/go-ucanto/core/result"
+	"github.com/storacha/go-ucanto/core/result/failure"
+	"github.com/storacha/go-ucanto/core/result/failure/datamodel"
+	"github.com/storacha/go-ucanto/principal"
+	serverdatamodel "github.com/storacha/go-ucanto/server/datamodel"
+	"github.com/storacha/guppy/pkg/delegation"
+)
+
+// ClaimAccess fetches any stored delegations from the service. This is the
+// second step of the Agent authorization process, from the Agent's point of
+// view. After the Agent has [RequestAccess]ed, the service will instruct the
+// user to confirm the access request out of band, e.g. via email. Once
+// confirmed, a delegation will be available on the service for the Agent to
+// claim.
+func ClaimAccess(issuer principal.Signer, options ...Option) ([]udelegation.Delegation, error) {
+	cfg, err := NewConfig(options...)
+	if err != nil {
+		return nil, err
+	}
+
+	caveats := access.ClaimCaveats{}
+
+	inv, err := access.Claim.Invoke(issuer, cfg.conn.ID(), issuer.DID().String(), caveats, convertToInvocationOptions(cfg)...)
+	if err != nil {
+		return nil, fmt.Errorf("generating invocation: %w", err)
+	}
+
+	resp, err := uclient.Execute([]invocation.Invocation{inv}, cfg.conn)
+	if err != nil {
+		return nil, fmt.Errorf("sending invocation: %w", err)
+	}
+
+	reader, err := receipt.NewReceiptReaderFromTypes[access.ClaimOk, serverdatamodel.HandlerExecutionErrorModel](access.ClaimOkType(), serverdatamodel.HandlerExecutionErrorType(), captypes.Converters...)
+	if err != nil {
+		return nil, fmt.Errorf("generating receipt reader: %w", err)
+	}
```

Comment:
in https://github.com/storacha/go-libstoracha/pull/32 we added convenience type definitions for receipts and receipt readers for a number of capabilities. Feel free to add them there to be used here.

On a second look of that other PR, I think the usability can be improved further, but that is something we can iterate on when we use it more.

Reply:
I'm going to put this off for a bit, but this is super helpful to know.